### DISCUSSION TITLE
FP script to zoom in a map of 5% (increase of 5% the view): need of help to correct

### DISCUSSION THREAD

**Discussion by Illegit** [freeplane/freeplane#1042] [Help] [answered]
Hi All, I ask if anybody can suggest how to correct the following script meant to zoom in on the map enlarging the map of 5%, it turns out with a null error:

import freemind.controller.MapController;

MapController controller = FreeplaneStarter.getController();
controller.getMapView().zoom(1.05);

Thanks for any suggestions.

**Comment by lilive** [comment]
Hi,
freemind.controller.MapController do not exists.
Is this a Freemind script ?
Do you try to achieve something special here, or do you just want a regular Freeplane script ?
In this case, you just need:
```
c.zoom = 1.05
```
to set the zoom to 105%
or
```
c.zoom = c.zoom * 1.05
```
to increase the zoom by 5%

**Comment by lilive** [comment, marked as answer]
You can also trigger the menu action `View > Zoom > Zoom In` with
```
menuUtils.executeMenuItems(['ZoomInAction'])
```

**Comment by Illegit** [comment]
I was totally wrong, the solution was much simpler. Many thanks it is working nicely.

**Comment by lilive** [comment]
Glad to help.
For your information:

To get details about a script error
`Menu Tools > Open user directory`, then open the file `logs/log.0` (for example with a text editor).
Read details about the error at the end of the file.

In this case, the log.0 contained 
```
unable to resolve class freemind.controller.MapController
 @ line 3, column 1.
   import freemind.controller.MapController;
   ^
```
meaning that there is no existing class freemind.controller.MapController

Most of the time, you will find the methods you need in the [Freeplane scripting API](https://docs.freeplane.org/scripting/Scripting.html#the-scripting-api).

In this case, `Help > Scripting API`, and type `zoom` in the search box lead to the `org.freeplane.api.Controller.setZoom()` method.

It is usefull to remember that we can trigger Freeplane menu items in groovy with `MenuUtils.executeMenuItems()`. Use [`Developer Tools > Menu item info`](https://docs.freeplane.org/scripting/Scripting.html#menu-locations) to find the key name for the entry we want.

**Comment by Illegit** [comment]
Thank you lilive. Appreciated. I'll follow your suggestions.
